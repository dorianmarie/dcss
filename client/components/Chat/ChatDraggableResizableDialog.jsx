import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Rnd } from 'react-rnd';
import { Ref } from '@components/UI';
import Layout from '@utils/Layout';

import './Chat.css';

const outerBaseClassNames =
  'ui modal transition visible active c__container-modal';
const outerMinClassName = `${outerBaseClassNames} c__minimized`;
const outerMaxClassName = `${outerBaseClassNames} resizable`;

const BASE_HEIGHT = 590;
const BASE_WIDTH = 430;

const BASE_X = 0;
const BASE_Y = 100;

const rndStyleUserSelectNone = {
  WebkitUserSelect: 'none',
  WebkitTouchCallout: 'none',
  MozUserSelect: 'none',
  msUserSelect: 'none',
  userSelect: 'none'
};

const rndStyleUserSelectAuto = {
  WebkitUserSelect: 'auto',
  WebkitTouchCallout: 'auto',
  MozUserSelect: 'auto',
  msUserSelect: 'auto',
  userSelect: 'auto'
};

class ChatDraggableResizableDialog extends Component {
  constructor(props) {
    super(props);

    const {
      dimensions: { height = BASE_HEIGHT, width = BASE_WIDTH },
      position: { x, y }
    } = this.props;

    const rndStyle = {};

    this.state = {
      width,
      height,
      x,
      y,
      rndStyle
    };

    this.ref = null;
  }

  componentDidMount() {
    const width = this.state.width || BASE_WIDTH;
    const height = this.state.height || BASE_HEIGHT;
    const x = this.state.x || BASE_X;
    const y = this.state.y || BASE_Y;
    this.setState({
      height,
      width,
      x,
      y
    });
  }

  render() {
    const { children, isMinimized } = this.props;

    const { height, width, x, y } = this.state;

    if (!width || !height) {
      return null;
    }

    let outerMinMaxClassName = isMinimized
      ? outerMinClassName
      : outerMaxClassName;

    let rndStyle = {
      ...this.state.rndStyle,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      border: 'solid 1px #ddd',
      background: '#f0f0f0'
    };

    if (isMinimized) {
      rndStyle = {
        border: 'unset',
        background: 'unset'
        // display: 'none'
      };
    }

    const onDragResizeStart = () => {
      this.setState({
        rndStyle: {
          ...rndStyleUserSelectNone
        }
      });

      if (this.props.onDragResizeStart) {
        this.props.onDragResizeStart();
      }
    };

    const onDragResizeStop = data => {
      this.setState({
        rndStyle: {
          ...rndStyleUserSelectAuto
        }
      });
      if (this.props.onDragResizeStop) {
        this.props.onDragResizeStop(data);
      }
    };

    const updateState = ({ height, width, x, y }) => {
      this.setState({ height, width, x, y });
      return {
        height,
        width,
        x,
        y
      };
    };

    const onDragStop = (e, { x, y }) => {
      onDragResizeStop(
        updateState({
          ...this.state,
          x,
          y
        })
      );
    };

    const onResize = (e, direction, resizer, delta, position) => {
      const height = Math.max(parseInt(resizer.style.height), BASE_HEIGHT);
      const width = Math.max(parseInt(resizer.style.width), BASE_WIDTH);
      const update = {
        height,
        width,
        ...position
      };
      this.setState(update);
      if (this.props.onDragResize) {
        this.props.onDragResize(update);
      }
    };

    const onDrag = (e, { x, y }) => {
      const update = {
        ...this.state,
        x,
        y
      };
      if (this.props.onDragResize) {
        this.props.onDragResize(update);
      }
    };

    const onResizeStop = (e, direction, resizer, delta, position) => {
      const height = Math.max(parseInt(resizer.style.height), BASE_HEIGHT);
      const width = Math.max(parseInt(resizer.style.width), BASE_WIDTH);
      onDragResizeStop(
        updateState({
          height,
          width,
          ...position
        })
      );
    };

    const dialogBox = (
      <Ref innerRef={node => (this.ref = node)}>
        <div role="dialog" className={outerMinMaxClassName}>
          {children}
        </div>
      </Ref>
    );

    return Layout.isNotForMobile() ? (
      <Rnd
        resizeHandleWrapperClass="c__size-handle"
        dragHandleClassName="c__drag-handle"
        minHeight={BASE_HEIGHT}
        minWidth={BASE_WIDTH}
        position={{ x, y }}
        size={{ width, height }}
        style={rndStyle}
        onDragStart={onDragResizeStart}
        onDragStop={onDragStop}
        onResizeStart={onDragResizeStart}
        onDrag={onDrag}
        onResize={onResize}
        onResizeStop={onResizeStop}
      >
        {dialogBox}
      </Rnd>
    ) : (
      dialogBox
    );
  }
}

ChatDraggableResizableDialog.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  dimensions: PropTypes.object,
  isMinimized: PropTypes.bool,
  onDragResizeStart: PropTypes.func,
  onDragResize: PropTypes.func,
  onDragResizeStop: PropTypes.func,
  position: PropTypes.object
};

ChatDraggableResizableDialog.defaultProps = {
  dimensions: {
    width: BASE_WIDTH,
    height: BASE_HEIGHT
  },
  position: {
    x: 0,
    y: 100
  }
};

export default ChatDraggableResizableDialog;
