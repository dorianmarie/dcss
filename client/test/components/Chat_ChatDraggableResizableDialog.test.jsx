/** @TEMPLATE: BEGIN **/
import React from 'react';
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect
}));

import {
  fetchImplementation,
  mounter,
  reduxer,
  serialize,
  snapshotter,
  state
} from '../bootstrap';
import { unmountComponentAtNode } from 'react-dom';

import {
  act,
  fireEvent,
  prettyDOM,
  render,
  screen,
  waitFor
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

async function waitForPopper() {
  // Popper update() - https://github.com/popperjs/react-popper/issues/350
  await act(async () => await null);
}

/** @TEMPLATE: END **/

/** @GENERATED: BEGIN **/

import Emitter from 'events';

import Storage from '@utils/Storage';
jest.mock('@utils/Storage', () => {
  return {
    __esModule: true,
    ...jest.requireActual('@utils/Storage')
  };
});

import Layout from '@utils/Layout';
jest.mock('@utils/Layout', () => {
  return {
    ...jest.requireActual('@utils/Layout'),
    isForMobile: jest.fn(() => false),
    isNotForMobile: jest.fn(() => true)
  };
});

globalThis.rndProps = {};
jest.mock('react-rnd', () => {
  function Rnd(props) {
    globalThis.rndProps = {
      ...props
    };

    return <div>{props.children}</div>;
  }

  return {
    __esModule: true,
    Rnd
  };
});

import ChatDraggableResizableDialog from '../../components/Chat/ChatDraggableResizableDialog.jsx';
/** @GENERATED: END **/

/** @TEMPLATE: BEGIN **/
const original = JSON.parse(JSON.stringify(state));
let container = null;
let commonProps = null;
let commonState = null;
/** @TEMPLATE: END **/

beforeAll(() => {
  /** @TEMPLATE: BEGIN **/
  (window || global).fetch = jest.fn();
  /** @TEMPLATE: END **/
});

afterAll(() => {
  /** @TEMPLATE: BEGIN **/
  jest.restoreAllMocks();
  /** @TEMPLATE: END **/
});

beforeEach(() => {
  /** @TEMPLATE: BEGIN **/
  jest.useFakeTimers();
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);
  /** @TEMPLATE: END **/

  /** @GENERATED: BEGIN **/

  globalThis.rndProps = null;
  jest.spyOn(window, 'addEventListener');
  jest.spyOn(window, 'removeEventListener');

  /** @GENERATED: END **/

  /** @TEMPLATE: BEGIN **/
  commonProps = {};
  commonState = JSON.parse(JSON.stringify(original));
  /** @TEMPLATE: END **/
});

afterEach(() => {
  /** @TEMPLATE: BEGIN **/
  jest.runOnlyPendingTimers();
  jest.useRealTimers();
  jest.resetAllMocks();
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  commonProps = null;
  commonState = null;
  /** @TEMPLATE: END **/
});

test('ChatDraggableResizableDialog', () => {
  expect(ChatDraggableResizableDialog).toBeDefined();
});

/** @GENERATED: BEGIN **/
test('Render 1 1', async done => {
  const Component = ChatDraggableResizableDialog;
  const props = {
    ...commonProps,
    dimensions: {},
    position: {},
    isMinimized: false,
    onDragResizeStart: jest.fn(),
    onDragResize: jest.fn(),
    onDragResizeStop: jest.fn()
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();

  done();
});
/** @GENERATED: END **/

/* INJECTION STARTS HERE */

test('isMinimized', async done => {
  const Component = ChatDraggableResizableDialog;

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    isMinimized: true,
    children
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  done();
});

test('isMinimized && Layout.isNotForMobile()', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    isMinimized: true,
    children
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  done();
});

test('Default height and width when undefined', async done => {
  const Component = ChatDraggableResizableDialog;

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  done();
});

test('Default height and width when 0', async done => {
  const Component = ChatDraggableResizableDialog;

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children,
    dimentions: {
      height: 0,
      width: 0
    }
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  done();
});

test('Rnd', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();

  expect(globalThis.rndProps).toMatchInlineSnapshot(`
    Object {
      "bounds": "window",
      "children": <Ref
        innerRef={[Function]}
      >
        <div
          className="ui modal transition visible active c__container-modal resizable"
          role="dialog"
        >
          <div>
            a
          </div>
        </div>
      </Ref>,
      "disableDragging": undefined,
      "dragHandleClassName": "c__drag-handle",
      "minHeight": 590,
      "minWidth": 456,
      "onDrag": [Function],
      "onDragStart": [Function],
      "onDragStop": [Function],
      "onResize": [Function],
      "onResizeStart": [Function],
      "onResizeStop": [Function],
      "position": Object {
        "x": 0,
        "y": 0,
      },
      "resizeHandleWrapperClass": "c__size-handle",
      "size": Object {
        "height": 590,
        "width": 456,
      },
      "style": Object {
        "alignItems": "center",
        "background": "#f0f0f0",
        "display": "flex",
        "justifyContent": "center",
      },
    }
  `);
  done();
});

test('No Rnd on mobile', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return false;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);
  done();
});

test('Rnd: onDragStop/onResizeStop', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children,
    dimensions: {},
    position: {},
    isMinimized: false,
    onDragResizeStop: jest.fn()
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  expect(globalThis.rndProps).toMatchInlineSnapshot(`
    Object {
      "bounds": "window",
      "children": <Ref
        innerRef={[Function]}
      >
        <div
          className="ui modal transition visible active c__container-modal resizable"
          role="dialog"
        >
          <div>
            a
          </div>
        </div>
      </Ref>,
      "disableDragging": false,
      "dragHandleClassName": "c__drag-handle",
      "minHeight": 590,
      "minWidth": 456,
      "onDrag": [Function],
      "onDragStart": [Function],
      "onDragStop": [Function],
      "onResize": [Function],
      "onResizeStart": [Function],
      "onResizeStop": [Function],
      "position": Object {
        "x": undefined,
        "y": undefined,
      },
      "resizeHandleWrapperClass": "c__size-handle",
      "size": Object {
        "height": 590,
        "width": 456,
      },
      "style": Object {
        "alignItems": "center",
        "background": "#f0f0f0",
        "display": "flex",
        "justifyContent": "center",
      },
    }
  `);

  const event = {};
  const direction = 'top';
  const resizer = document.createElement('div');
  const delta = { width: 400, height: 400 };
  const position = { x: 393, y: 174 };
  // e, { x, y }
  globalThis.rndProps.onDragStop(event, { ...delta, ...position });

  // e, direction, resizer, delta, position
  globalThis.rndProps.onResizeStop(event, direction, resizer, delta, position);

  await waitFor(() => expect(props.onDragResizeStop).toHaveBeenCalledTimes(2));

  expect(props.onDragResizeStop.mock.calls[0]).toMatchInlineSnapshot(`
    Array [
      Object {
        "height": 590,
        "width": 456,
        "x": 393,
        "y": 174,
      },
    ]
  `);
  done();
});

test('Rnd: onDragStop/onResizeStop without props.onDragResizeStop', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children,
    dimensions: {},
    position: {},
    isMinimized: false
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  expect(globalThis.rndProps).toMatchInlineSnapshot(`
    Object {
      "bounds": "window",
      "children": <Ref
        innerRef={[Function]}
      >
        <div
          className="ui modal transition visible active c__container-modal resizable"
          role="dialog"
        >
          <div>
            a
          </div>
        </div>
      </Ref>,
      "disableDragging": false,
      "dragHandleClassName": "c__drag-handle",
      "minHeight": 590,
      "minWidth": 456,
      "onDrag": [Function],
      "onDragStart": [Function],
      "onDragStop": [Function],
      "onResize": [Function],
      "onResizeStart": [Function],
      "onResizeStop": [Function],
      "position": Object {
        "x": undefined,
        "y": undefined,
      },
      "resizeHandleWrapperClass": "c__size-handle",
      "size": Object {
        "height": 590,
        "width": 456,
      },
      "style": Object {
        "alignItems": "center",
        "background": "#f0f0f0",
        "display": "flex",
        "justifyContent": "center",
      },
    }
  `);

  const event = {};
  const direction = 'top';
  const resizer = document.createElement('div');
  const delta = { width: 400, height: 400 };
  const position = { x: 393, y: 174 };
  // e, { x, y }
  globalThis.rndProps.onDragStop(event, { ...delta, ...position });

  // e, direction, resizer, delta, position
  globalThis.rndProps.onResizeStop(event, direction, resizer, delta, position);

  expect(serialize()).toMatchSnapshot();
  done();
});

test('Rnd: onDrag/onResize', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children,
    dimensions: {},
    position: {},
    isMinimized: false,
    onDragResize: jest.fn()
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  expect(globalThis.rndProps).toMatchInlineSnapshot(`
    Object {
      "bounds": "window",
      "children": <Ref
        innerRef={[Function]}
      >
        <div
          className="ui modal transition visible active c__container-modal resizable"
          role="dialog"
        >
          <div>
            a
          </div>
        </div>
      </Ref>,
      "disableDragging": false,
      "dragHandleClassName": "c__drag-handle",
      "minHeight": 590,
      "minWidth": 456,
      "onDrag": [Function],
      "onDragStart": [Function],
      "onDragStop": [Function],
      "onResize": [Function],
      "onResizeStart": [Function],
      "onResizeStop": [Function],
      "position": Object {
        "x": undefined,
        "y": undefined,
      },
      "resizeHandleWrapperClass": "c__size-handle",
      "size": Object {
        "height": 590,
        "width": 456,
      },
      "style": Object {
        "alignItems": "center",
        "background": "#f0f0f0",
        "display": "flex",
        "justifyContent": "center",
      },
    }
  `);

  const event = {};
  const direction = 'top';
  const resizer = document.createElement('div');
  const delta = { width: 400, height: 400 };
  const position = { x: 393, y: 174 };
  // e, { x, y }
  globalThis.rndProps.onDrag(event, { ...delta, ...position });

  // e, direction, resizer, delta, position
  globalThis.rndProps.onResize(event, direction, resizer, delta, position);

  await waitFor(() => expect(props.onDragResize).toHaveBeenCalledTimes(2));

  expect(props.onDragResize.mock.calls[0]).toMatchInlineSnapshot(`
    Array [
      Object {
        "height": 590,
        "rndStyle": Object {},
        "width": 456,
        "x": 393,
        "y": 174,
      },
    ]
  `);
  done();
});

test('Rnd: onDrag/onResize without props.onDragResize', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children,
    dimensions: {},
    position: {},
    isMinimized: false
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  expect(globalThis.rndProps).toMatchInlineSnapshot(`
    Object {
      "bounds": "window",
      "children": <Ref
        innerRef={[Function]}
      >
        <div
          className="ui modal transition visible active c__container-modal resizable"
          role="dialog"
        >
          <div>
            a
          </div>
        </div>
      </Ref>,
      "disableDragging": false,
      "dragHandleClassName": "c__drag-handle",
      "minHeight": 590,
      "minWidth": 456,
      "onDrag": [Function],
      "onDragStart": [Function],
      "onDragStop": [Function],
      "onResize": [Function],
      "onResizeStart": [Function],
      "onResizeStop": [Function],
      "position": Object {
        "x": undefined,
        "y": undefined,
      },
      "resizeHandleWrapperClass": "c__size-handle",
      "size": Object {
        "height": 590,
        "width": 456,
      },
      "style": Object {
        "alignItems": "center",
        "background": "#f0f0f0",
        "display": "flex",
        "justifyContent": "center",
      },
    }
  `);

  const event = {};
  const direction = 'top';
  const resizer = document.createElement('div');
  const delta = { width: 400, height: 400 };
  const position = { x: 393, y: 174 };
  // e, { x, y }
  globalThis.rndProps.onDrag(event, { ...delta, ...position });

  // e, direction, resizer, delta, position
  globalThis.rndProps.onResize(event, direction, resizer, delta, position);

  expect(serialize()).toMatchSnapshot();
  done();
});

test('Rnd: onDragStart/onResizeStart', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children,
    dimensions: {},
    position: {},
    isMinimized: false,
    onDragResizeStart: jest.fn()
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  expect(globalThis.rndProps).toMatchInlineSnapshot(`
    Object {
      "bounds": "window",
      "children": <Ref
        innerRef={[Function]}
      >
        <div
          className="ui modal transition visible active c__container-modal resizable"
          role="dialog"
        >
          <div>
            a
          </div>
        </div>
      </Ref>,
      "disableDragging": false,
      "dragHandleClassName": "c__drag-handle",
      "minHeight": 590,
      "minWidth": 456,
      "onDrag": [Function],
      "onDragStart": [Function],
      "onDragStop": [Function],
      "onResize": [Function],
      "onResizeStart": [Function],
      "onResizeStop": [Function],
      "position": Object {
        "x": undefined,
        "y": undefined,
      },
      "resizeHandleWrapperClass": "c__size-handle",
      "size": Object {
        "height": 590,
        "width": 456,
      },
      "style": Object {
        "alignItems": "center",
        "background": "#f0f0f0",
        "display": "flex",
        "justifyContent": "center",
      },
    }
  `);

  const event = {};
  const direction = 'top';
  const resizer = document.createElement('div');
  const delta = { width: 400, height: 400 };
  const position = { x: 393, y: 174 };
  // e, { x, y }
  globalThis.rndProps.onDragStart(event, { ...delta, ...position });

  // e, direction, resizer, delta, position
  globalThis.rndProps.onResizeStart(event, direction, resizer, delta, position);

  await waitFor(() => expect(props.onDragResizeStart).toHaveBeenCalledTimes(2));

  expect(props.onDragResizeStart.mock.calls[0]).toMatchInlineSnapshot(
    `Array []`
  );
  done();
});

test('Rnd: onDragStart/onResizeStart without props.onDragResizeStart', async done => {
  const Component = ChatDraggableResizableDialog;

  Layout.isNotForMobile.mockImplementation(() => {
    return true;
  });

  const children = <div>a</div>;

  const props = {
    ...commonProps,
    children,
    dimensions: {},
    position: {},
    isMinimized: false
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  expect(globalThis.rndProps).toMatchInlineSnapshot(`null`);

  await render(<ConnectedRoutedComponent {...props} />);
  expect(serialize()).toMatchSnapshot();
  expect(globalThis.rndProps).toMatchInlineSnapshot(`
    Object {
      "bounds": "window",
      "children": <Ref
        innerRef={[Function]}
      >
        <div
          className="ui modal transition visible active c__container-modal resizable"
          role="dialog"
        >
          <div>
            a
          </div>
        </div>
      </Ref>,
      "disableDragging": false,
      "dragHandleClassName": "c__drag-handle",
      "minHeight": 590,
      "minWidth": 456,
      "onDrag": [Function],
      "onDragStart": [Function],
      "onDragStop": [Function],
      "onResize": [Function],
      "onResizeStart": [Function],
      "onResizeStop": [Function],
      "position": Object {
        "x": undefined,
        "y": undefined,
      },
      "resizeHandleWrapperClass": "c__size-handle",
      "size": Object {
        "height": 590,
        "width": 456,
      },
      "style": Object {
        "alignItems": "center",
        "background": "#f0f0f0",
        "display": "flex",
        "justifyContent": "center",
      },
    }
  `);

  const event = {};
  const direction = 'top';
  const resizer = document.createElement('div');
  const delta = { width: 400, height: 400 };
  const position = { x: 393, y: 174 };
  // e, { x, y }
  globalThis.rndProps.onDragStart(event, { ...delta, ...position });

  // e, direction, resizer, delta, position
  globalThis.rndProps.onResizeStart(event, direction, resizer, delta, position);

  expect(serialize()).toMatchSnapshot();
  done();
});
