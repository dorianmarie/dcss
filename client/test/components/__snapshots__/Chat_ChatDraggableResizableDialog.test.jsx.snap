// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default height and width when 0 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div
      class=\\"ui modal transition visible active c__container-modal resizable\\"
      role=\\"dialog\\"
      style=\\"position: absolute;\\"
    >
      <div>
        a
      </div>
    </div>
  </div>
</body>"
`;

exports[`Default height and width when undefined 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div
      class=\\"ui modal transition visible active c__container-modal resizable\\"
      role=\\"dialog\\"
      style=\\"position: absolute;\\"
    >
      <div>
        a
      </div>
    </div>
  </div>
</body>"
`;

exports[`No Rnd on mobile 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div
      class=\\"ui modal transition visible active c__container-modal resizable\\"
      role=\\"dialog\\"
      style=\\"position: absolute;\\"
    >
      <div>
        a
      </div>
    </div>
  </div>
</body>"
`;

exports[`Render 1 1 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div
      class=\\"ui modal transition visible active c__container-modal resizable\\"
      role=\\"dialog\\"
      style=\\"position: absolute;\\"
    />
  </div>
</body>"
`;

exports[`Rnd 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDrag/onResize 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDrag/onResize without props.onDragResize 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDrag/onResize without props.onDragResize 4`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div />
</body>"
`;

exports[`Rnd: onDragStart/onResizeStart 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDragStart/onResizeStart without props.onDragResizeStart 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDragStart/onResizeStart without props.onDragResizeStart 4`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDragStop/onResizeStop 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDragStop/onResizeStop without props.onDragResizeStop 2`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"ui modal transition visible active c__container-modal resizable\\"
        role=\\"dialog\\"
      >
        <div>
          a
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Rnd: onDragStop/onResizeStop without props.onDragResizeStop 4`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div />
</body>"
`;
